server/routerlicious/packages/routerlicious-base/src/utils/sessionHelper.ts

```diff
@@ -287,7 +288,35 @@ export async function getSession(
 ): Promise<ISession> {
 	const baseLumberjackProperties = getLumberBaseProperties(documentId, tenantId);
 
-	const document = await documentRepository.readOne({ tenantId, documentId });
+	let document;
```

Comment:
IMO, cleaner and more consistent with the rest of the logic in this codeblock to do 
```ts
const document = await runWithRetry(...).catch((error) => {
  // log error
  if (isNetworkError(error) && error.code === 404) {
    return undefined;
  }
  throw error;
});
```
Then, the `undefined` will be handled by the below condition that throws a 404 without `canRetry=true`, which would be correct if the retries all fail

Reply:
Fixed